var/spack/repos/builtin/packages/esmf/package.py

```diff
@@ -270,13 +270,18 @@ def setup_build_environment(self, env):
         elif self.pkg.compiler.name == "intel" or self.pkg.compiler.name == "oneapi":
             env.set("ESMF_COMPILER", "intel")
         elif self.pkg.compiler.name in ["clang", "apple-clang"]:
-            env.set("ESMF_COMPILER", "gfortranclang")
-            with self.pkg.compiler.compiler_environment():
-                gfortran_major_version = int(
-                    spack.compiler.get_compiler_version_output(
-                        self.pkg.compiler.fc, "-dumpversion"
-                    ).split(".")[0]
-                )
+            if "flang" in self.pkg.compiler.fc:
+                env.set("ESMF_COMPILER", "llvm")
+            elif "gfortran" in self.pkg.compiler.fc:
```

Comment:
This is fine for now. When #45189 will be merged, checks like this one could be written as:
```python
if self.spec["fortran"].name == "llvm":
    # Flang case
elif self.spec["fortran"].name == "gcc":
    # gfortran case
```

Reply:
Oh, I like that! Thanks @alalazo.